Visual glitch when updating the data for pivot table visualization

### Describe the bug

The screen turns blank when changing the query while the pivot table visualization is selected.

### To Reproduce

1. Open "Feedback" table from Sample Database
2. Choose pivot table as the visualization without aggregating the query
3. You should see this error
![Image](https://github.com/user-attachments/assets/f2b1ee6a-0e2a-4e55-95c0-fa7d00c2685e)
4. Click on "Summarize" - the "Count" will be preselected
5. Click "Done" in the summarize sidebar
6. New play icon should appear - click on it to run the updated query
7. The page turns blank

No errors in the console.
Only a duplicated warning (see logs below).


### Expected behavior

_No response_

### Logs

```
settings.js:107 Error getting setting pivot_table.column_split TypeError: Cannot read properties of undefined (reading 'name')
    at settings.ts:130:33
    at Array.map (<anonymous>)
    at settings.ts:130:16
    at ec.eV [as mapObject] (mapObject.js:13:27)
    at Object.getValue (settings.ts:129:19)
    at e (settings.js:84:56)
    at e (settings.js:66:5)
    at A (settings.js:110:1)
    at f (visualization.js:97:10)
    at selectors.js:732:25
```


### Information about your Metabase installation

```JSON
local dev, master @9cd41f0b6f3d33a9ccd7460252db105f6e4ab902
```

### Severity

p3 - p2

### Additional context

_No response_

--- a/frontend/src/metabase/visualizations/visualizations/PivotTable/settings.ts
+++ b/frontend/src/metabase/visualizations/visualizations/PivotTable/settings.ts
@@ -125,7 +125,8 @@ export const settings = {
 
         if (dimensions.length < 2) {
           columns = [];
-          rows = [first];
+          // use `dimensions` so `rows` is `[]` when there are zero dimensions, not `[undefined]` (metabase#56235)
+          rows = dimensions;
         } else if (dimensions.length <= 3) {
           columns = [first];
           rows = [second, ...rest];

--- a/frontend/src/metabase/visualizations/visualizations/PivotTable/settings.unit.spec.ts
+++ b/frontend/src/metabase/visualizations/visualizations/PivotTable/settings.unit.spec.ts
@@ -1,0 +1,36 @@
+import {
+  createMockCard,
+  createMockColumn,
+  createMockDatasetData,
+} from "metabase-types/api/mocks";
+
+import { settings } from "./settings";
+
+const COLUMN_SPLIT_SETTING = "pivot_table.column_split";
+
+describe("PivotTable settings", () => {
+  describe(`${COLUMN_SPLIT_SETTING} getValue`, () => {
+    it("should not throw when the query has no dimension columns (metabase#56235)", () => {
+      // a Count-only query has zero dimensions, which previously left an
+      // `undefined` slot in `rows` and threw when we mapped to `col.name`
+      const data = createMockDatasetData({
+        rows: [[0]],
+        cols: [
+          createMockColumn({
+            name: "count",
+            display_name: "Count",
+            source: "aggregation",
+            base_type: "type/Integer",
+            effective_type: "type/Integer",
+          }),
+        ],
+      });
+
+      const card = createMockCard({ display: "pivot" });
+
+      expect(() => {
+        settings[COLUMN_SPLIT_SETTING].getValue([{ data, card }], {});
+      }).not.toThrow();
+    });
+  });
+});

